python/cudf/cudf/core/dataframe.py

```diff
@@ -2633,6 +2633,73 @@ def insert(self, loc, name, value):
 
         self._data.insert(name, value, loc=loc)
 
+    def diff(self, periods=1, axis=0):
+        """
+        Calculates the difference of a Dataframe element compared with
+        another element in the Dataframe, treating each column independently
+
+        Parameters
+        ----------
+        periods : int, default 1
+            Periods to shift for calculating difference,
+            accepts negative values.
+        axis : {0 or ‘index’, 1 or ‘columns’}, default 0
+            Take difference over rows (0) or columns (1).
+            Only row-wise (0) shift is supported.
+
+        Returns
+        -------
+        DataFrame
+            First differences of the DataFrame.
+
+        Notes
+        -----
+        Diff currently only supports float and integer dtype columns
+
+        Examples
+        --------
+        >>> import cudf
+        >>> gdf = cudf.DataFrame({'a': [1, 2, 3, 4, 5, 6],
+                                  'b': [1, 1, 2, 3, 5, 8],
+                                  'c': [1, 4, 9, 16, 25, 36]})
+        >>> gdf
+        a  b   c
+        0  1  1   1
+        1  2  1   4
+        2  3  2   9
+        3  4  3  16
+        4  5  5  25
+        5  6  8  36
+        >>> gdf.diff(periods=2)
+            a     b     c
+        0  <NA>  <NA>  <NA>
+        1  <NA>  <NA>  <NA>
+        2     2     1     8
+        3     2     2    12
+        4     2     3    16
+        5     2     5    20
+
+        """
+
+        if not axis == 0:
+            raise NotImplementedError("Only axis=0 is supported.")
+
+        if not (np.issubdtype(i, np.number) for i in self.dtypes):
```

Comment:
I think this might be missing an `all`? Is this meant to be the following?

```python
 if not all(np.issubdtype(i, np.number) for i in self.dtypes):
```

In any event, we should prefer our cudf internal utilities for determining if a dtype is numeric rather than numpy machinery in general:

https://github.com/rapidsai/cudf/blob/branch-22.02/python/cudf/cudf/api/types.py

you could use `is_numeric_dtype` or `_is_non_decimal_numeric_dtype` to exclude decimals.